### PR TITLE
refactor: Clean separation between prompting and writing in adp-tooling

### PR DESCRIPTION
**Pull request by mmilko01**
https://github.com/SAP/open-ux-tools/issues/2216

### CODE REVIEW

**Review by tobiasqueck — DISMISSED** [review]
* changeset ok
* refactored code nicely separates prompting from writing
* locally tested some of the touched functionality without problems

File: packages/adp-tooling/src/base/change-utils.ts [hunk 1]
```diff
@@ -39,15 +39,15 @@ export function writeAnnotationChange(
     fs: Editor
 ): void {
     try {
-        const { fileName, answers } = data;
+        const { annotation } = data;
```
Comment (testojs) [inline packages/adp-tooling/src/base/change-utils.ts:42]: The `data` parameter of this function is of type `AnnotationsData`. This means that the function requires  also`variant` to be provided, but it is never used. 
Here we see that only the `annotation` object is used from the `data` parameter.

**Review by testojs — CHANGES_REQUESTED** [review]

File: packages/adp-tooling/src/base/change-utils.ts (hunk 1 above)
Comment (tobiasqueck) [reply packages/adp-tooling/src/base/change-utils.ts:42]: Changing the signature to:
```
export function writeAnnotationChange(
    projectPath: string,
    timestamp: number,
    { annotation }: AnnotationsData,
    change: ManifestChangeProperties,
    fs: Editor
): void {
```
should work (I think).

Comment (mmilko01) [reply packages/adp-tooling/src/base/change-utils.ts]: I tried to change the type and use `Pick` to get only the `annotation` part, but as the properties are nested, it became quite ugly, so I followed Tobias' advice :) 

Fixed with: eb8529ef6a8aa2dc6f1d959c41e796f5e2d56d85

Comment (testojs) [reply packages/adp-tooling/src/base/change-utils.ts]: But with this solution, the function still expects the whole object, including `variant`, to be provided.

**Review by testojs — APPROVED** [review]
* code looks good
* comments addressed
* changeset OK
* test coverage is good

File: packages/adp-tooling/src/base/change-utils.ts (hunk 1 above)
Comment (tobiasqueck) [reply packages/adp-tooling/src/base/change-utils.ts]: Very nice.

**Review by tobiasqueck — APPROVED** [review]
reapproving

File: packages/adp-tooling/src/base/change-utils.ts (hunk 1 above)
Comment (testojs) [reply packages/adp-tooling/src/base/change-utils.ts]: Great, thanks!

**Pull request merged**